File: android/app/src/main/java/com/meta/wearable/dat/externalsampleapps/cameraaccess/data/UserProfile.kt
package com.meta.wearable.dat.externalsampleapps.cameraaccess.data

import androidx.room.Entity
import androidx.room.PrimaryKey

@Entity(tableName = "user_profile")
data class UserProfile(
    @PrimaryKey(autoGenerate = true)
    val id: Long = 0,
    val name: String = "",
    val gender: String = "", // "Male", "Female", "Other"
    val age: Int = 0,
    val heightCm: Int = 0,
    val weightKg: Int = 0,
    val fitnessGoal: String = "", // "build_muscle", "lose_weight", "stay_active", "improve_health"
    val experienceLevel: String = "", // "beginner", "intermediate", "advanced"
    val weeklyWorkouts: String = "", // "0-2", "3-5", "6+"
    val mirrorPhotoPath: String? = null,
    val createdAt: Long = System.currentTimeMillis(),
    val updatedAt: Long = System.currentTimeMillis()
)

File: ios/GymBro/Data/Models/UserProfile.swift
import Foundation
import SwiftData

@Model
final class UserProfile {
    var name: String
    var gender: String           // "Male", "Female", "Other"
    var age: Int
    var heightCm: Int
    var weightKg: Int
    var fitnessGoal: String      // "build_muscle", "lose_weight", "stay_active", "improve_health"
    var experienceLevel: String  // "beginner", "intermediate", "advanced"
    var weeklyWorkouts: String   // "0-2", "3-5", "6+"
    var mirrorPhotoPath: String?
    var createdAt: Date
    var updatedAt: Date

    init(
        name: String = "",
        gender: String = "",
        age: Int = 0,
        heightCm: Int = 0,
        weightKg: Int = 0,
        fitnessGoal: String = "",
        experienceLevel: String = "",
        weeklyWorkouts: String = "",
        mirrorPhotoPath: String? = nil,
        createdAt: Date = Date(),
        updatedAt: Date = Date()
    ) {
        self.name = name
        self.gender = gender
        self.age = age
        self.heightCm = heightCm
        self.weightKg = weightKg
        self.fitnessGoal = fitnessGoal
        self.experienceLevel = experienceLevel
        self.weeklyWorkouts = weeklyWorkouts
        self.mirrorPhotoPath = mirrorPhotoPath
        self.createdAt = createdAt
        self.updatedAt = updatedAt
    }
}
